crates/oxc_transformer/src/vite/module_runner_transform.rs

```diff
@@ -0,0 +1,2006 @@
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('vue', { importedNames: ['fn'] });
+function A() {
+  let { fn, test } = {
+    fn: 'foo',
+    test: 'bar'
+  };
+  return { fn };
+}",
+        );
+    }
+
+    #[test]
+    fn do_not_rewrite_destructuring_array() {
+        test_same(
+            "import { fn } from 'vue';function A(){ let [fn, test] = ['foo', 'bar']; return { fn }; }",
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('vue', { importedNames: ['fn'] });
+function A() {
+  let [fn, test] = ['foo', 'bar'];
+  return { fn };
+}",
+        );
+    }
+
+    #[test]
+    fn do_not_rewrite_catch_clause() {
+        test_same(
+            "import {error} from './dependency';try {} catch(error) {}",
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('./dependency', { importedNames: ['error'] });
+try {} catch (error) {}",
+        );
+    }
+
+    #[test]
+    fn should_declare_imported_super_class() {
+        test_same(
+            "import { Foo } from './dependency'; class A extends Foo {}",
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('./dependency', { importedNames: ['Foo'] });
+class A extends __vite_ssr_import_0__.Foo {}",
+        );
+    }
+
+    #[test]
+    fn should_declare_exported_super_class() {
+        test_same(
+            "import { Foo } from './dependency'; export default class A extends Foo {} export class B extends Foo {}",
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('./dependency', { importedNames: ['Foo'] });
+class A extends __vite_ssr_import_0__.Foo {}
+Object.defineProperty(__vite_ssr_exports__, 'default', {
+  enumerable: true,
+  configurable: true,
+  get() {
+    return A;
+  }
+});
+class B extends __vite_ssr_import_0__.Foo {}
+Object.defineProperty(__vite_ssr_exports__, 'B', {
+  enumerable: true,
+  configurable: true,
+  get() {
+    return B;
+  }
+});",
+        );
+    }
+
+    /// <https://github.com/vitejs/vite/issues/4049>
+    #[test]
+    fn should_handle_default_export_variants() {
+        // default anonymous functions
+        test_same(
+            "export default function() {}",
+            "Object.defineProperty(__vite_ssr_exports__, 'default', {
+  enumerable: true,
+  configurable: true,
+  get() {
+    return function() {};
+  }
+});",
+        );
+
+        // default anonymous class
+        test_same(
+            "export default class {}",
+            "Object.defineProperty(__vite_ssr_exports__, 'default', {
+  enumerable: true,
+  configurable: true,
+  get() {
+    return class {};
+  }
+});",
+        );
+
+        // default named functions
+        test_same(
+            "export default function foo() {}\nfoo.prototype = Object.prototype;",
+            "function foo() {}
+Object.defineProperty(__vite_ssr_exports__, 'default', {
+  enumerable: true,
+  configurable: true,
+  get() {
+    return foo;
+  }
+});
+foo.prototype = Object.prototype;",
+        );
```

Comment:
A little difference in this case, the `Object.defineProperty(__vite_ssr_exports__` is inserted following original export declaration but in Vite, it is inserted following `foo.prototype = Object.prototype;`

See https://github.com/vitejs/vite/blob/00deea4ff88e30e299cb40a801b5dc0205ac913d/packages/vite/src/node/ssr/__tests__/ssrTransform.spec.ts#L504-L514

I think it doesn't matter because there is no runtime behavior change if I understand correctly.